notebooks/106-speech-recognition-quantize-wav2vec2/106-speech-recognition-quantize-wav2vec2.ipynb

```diff
@@ -0,0 +1,631 @@
+    "        (index, annotation), data, metadata\n",
+    "        \"\"\"\n",
+    "        return self._ds[index]\n",
+    "\n",
+    "    def _read_flac(self, file_name):\n",
+    "        speech, samplerate = sf.read(file_name)\n",
+    "        assert samplerate == 16000, \"read_flac: only 16kHz supported!\"\n",
+    "        return speech.reshape(-1, 1).T\n",
+    "\n",
+    "    # Methods specific to the current implementation\n",
+    "    def _prepare_dataset(self):\n",
+    "        pattern = re.compile(r'([0-9\\-]+)\\s+(.+)')\n",
+    "        data_folder = Path(self._data_dir)\n",
+    "        txts = list(data_folder.glob('**/*.txt'))\n",
+    "        counter = 0\n",
+    "        for txt in txts:\n",
+    "            content = txt.open().readlines()\n",
+    "            for line in content:\n",
+    "                res = pattern.search(line)\n",
+    "                if not res: continue\n",
+    "                name = res.group(1)\n",
+    "                transcript = res.group(2)\n",
+    "                fname = txt.parent / name\n",
+    "                fname = fname.with_suffix('.flac')\n",
+    "                identifier = str(fname.relative_to(data_folder))\n",
+    "                self._ds.append(((counter, transcript.upper()), self._read_flac(os.path.join(self._data_dir, identifier))))\n",
+    "                counter += 1\n",
+    "                if counter >= self.samples_limit:\n",
+    "                    # Limit exceeded\n",
+    "                    return"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "71c867d3",
+   "metadata": {
+    "id": "Re9-YhbBddh3"
+   },
+   "source": [
+    "## Define WER Metric Calculation\n",
+    "At this step the `Metric` interface for WER metric is implemented. It is used for validating the accuracy of the model. `WER` stands for Word Error Rate, you can find more details on the [Wiki page](https://en.wikipedia.org/wiki/Word_error_rate)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2f4e7fbc",
+   "metadata": {
+    "id": "GB8L492ztZEC"
+   },
+   "outputs": [],
+   "source": [
+    "class MetricWER(Metric):\n",
+    "    alphabet = [\n",
+    "        \"<pad>\", \"<s>\", \"</s>\", \"<unk>\", \"|\",\n",
+    "        \"e\", \"t\", \"a\", \"o\", \"n\", \"i\", \"h\", \"s\", \"r\", \"d\", \"l\", \"u\",\n",
+    "        \"m\", \"w\", \"c\", \"f\", \"g\", \"y\", \"p\", \"b\", \"v\", \"k\", \"'\", \"x\", \"j\", \"q\", \"z\"]\n",
+    "    words_delimiter = '|'\n",
+    "    pad_token = '<pad>'\n",
+    "\n",
+    "    # Required methods\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self._name = \"WER\"\n",
+    "        self._sum_score = 0\n",
+    "        self._sum_words = 0\n",
+    "        self._cur_score = 0\n",
+    "        self._decoding_vocab = dict(enumerate(self.alphabet))\n",
+    "\n",
+    "    @property\n",
+    "    def value(self):\n",
```

Comment:
Do we really need to define `value()` here? It's needed only for the Accuracy Aware quantization algorithm.

Reply:
Unfortunately, we need, API makes is mandatory: https://github.com/openvinotoolkit/openvino/blob/master/tools/pot/openvino/tools/pot/api/metric.py